internal/apis/config/controller/fuzzer/fuzzer.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fuzzer
+
+import (
+	fuzz "github.com/google/gofuzz"
+	runtimeserializer "k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/utils/pointer"
+
+	"github.com/cert-manager/cert-manager/internal/apis/config/controller"
+)
+
+// Funcs returns the fuzzer functions for the controller config api group.
+var Funcs = func(codecs runtimeserializer.CodecFactory) []interface{} {
```

Comment:
@munnerz I'm having a bit of a hard time understanding what to do with this bit of code. When I run the [install/roundtrip_test.go](https://github.com/cert-manager/cert-manager/blob/357829f9391537e5d13ccdc5b5837b0b812d42e3/internal/apis/config/controller/install/roundtrip_test.go), I always seem to get a random failure. See below: 
```
roundtrip.go:254: 	round tripping to controller.config.cert-manager.io/v1alpha1, Kind=ControllerConfiguration v1alpha1.ControllerConfiguration
    roundtrip.go:361: ControllerConfiguration: diff:   &controller.ControllerConfiguration{
          	... // 23 identical fields
          	DefaultIssuerKind:                 "ƪ簡xȰ滣m3瞹)觲ǜDfǺĕh",
          	DefaultIssuerGroup:                "蹐Ä檄ȢxƲ鲗±",
        - 	DefaultAutoCertificateAnnotations: []string{},
        + 	DefaultAutoCertificateAnnotations: []string{"kubernetes.io/tls-acme"},
          	DNS01RecursiveNameservers:         {"8.8.8.8:53"},
          	DNS01RecursiveNameserversOnly:     &true,
          	... // 4 identical fields
          	EnablePprof:           &true,
          	DNS01CheckRetryPeriod: s"-125575h39m12.764419686s",
          	CopiedAnnotationPrefixes: []string{
        + 		"*",
        + 		"-kubectl.kubernetes.io/",
        + 		"-fluxcd.io/",
        + 		"-argocd.argoproj.io/",
          	},
          	FeatureGates: {"feature": true},
          }
```

 Is the problem that I am defaulting too many values in the API? I was trying to follow what you did with the Webhook implementation of this work, and I could use a bit of your input on what I've done so far.

Reply:
@AcidLeroy I think you want to ensure that if the fuzzer has passed an 'empty' value for any of the fields that you default here https://github.com/cert-manager/cert-manager/pull/5337/commits/cf9b3f9345e9010be8ac2beb235fad11b5727a7b#diff-5dabcfe938566d3a84f3f414ff948f3e5e64a3412d7bd64a6f287b3c2d160dc8R166-R300 then you override it here https://github.com/cert-manager/cert-manager/pull/5337/commits/cf9b3f9345e9010be8ac2beb235fad11b5727a7b#diff-011c4647b8e5840f40c424ff9262b55ccacbf9c30de180be46f74dab328c3ba8R29-R88 - I think you already have majority, but some are missing and for some the comparison is not correct - it should be the same as in defaulting code. The roundtrip test triggers the defaulting, so if we allow empty values in input, the output will be different because it will have the defaulted values.

Reply:
@AcidLeroy they don't have to be the same as defaults, no- the idea is that you don't leave them empty by setting some random value to prevent the defaulting from kicking in and changing what was passed.